lms/djangoapps/instructor/enrollment.py

```diff
@@ -267,6 +269,15 @@ def reset_student_attempts(course_id, student, module_state_key, requesting_user
 
     if delete_module:
         module_to_reset.delete()
+        if block and block.has_score:
```

Comment:
@efischer19 good point. I tried to delete student state for a non-graded block on stage, and got an error. Turns out trying to reset the state of a non-scored block already causes an error, and we handle it with the message "Failed to delete student state. Unknown Error Occurred." 

The error is caused by trying to get a non-existent StudentModule: https://github.com/edx/edx-platform/blob/master/lms/djangoapps/instructor/enrollment.py#L262

That means the code to fire `SCORE_CHANGED` won't ever even be reached in the case I described. I'm going to just get rid of the code that restricts firing `SCORE_CHANGED` to only blocks with a score.